src/Modal/Portal.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import cx from 'classnames';
+import Panel from '../Panel/Panel';
+import styles from "./Modal.module.scss";
+
+class Portal extends React.Component {
```

Comment:
No need for this indirection. All the corresponding logic to modal should be in `Modal` otherwise readability is decreased (indirections and a lot of duplicated code). `handleOnKeyUp` and mounting to portal should be in `Modal`